src/js/plugin/sdk/pipeline/PipelineModel.js

```diff
@@ -0,0 +1,48 @@
+
+class PipelineModel {
+
+  constructor() {
+    this.pipelines = new Map();
+  };
+
+  clearPipeline() {
+    this.pipelines.clear();
+  };
+
+  applyFilter(operationName, filterParameters) {
+    const pipeline = this.getPipeline(operationName);
+
+    pipeline.forEach((filter) => {
+      filter(filterParameters);
+    });
+    return filterParameters;
+  };
+
+  register(operationName, callback) {
+
+    const pipeline = this.getPipeline(operationName);
+
+    if (pipeline.indexOf(callback) === -1) {
+      pipeline.push(callback);
+    }
+  };
+
+  deregister(operationName, callback) {
+    const pipeline = this.getPipeline(operationName);
+
+    var callbckPosition = pipeline.indexOf(callback);
+    if (callbckPosition !== -1) {
+      pipeline.splice(callbckPosition, 1);
+    }
+  };
+
+  getPipeline(pipelineName) {
```

Comment:
I wouldn't expose this method as theoretically allows unwanted mutations. I'd also argue that as not necessary to extract the respective functionality as we only need to get the pipelines in a very few places and only in one place we want to actually create a new pipeline if there is none.